torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -580,6 +580,55 @@ RegisterOperators reg(
            return 0;
          },
          aliasAnalysisFromSchema()),
+     Operator(
+         "prim::is_leaf(Tensor a) -> bool",
+         [](Stack& stack) {
+           at::Tensor a;
+           pop(stack, a);
+           push(stack, a.is_leaf());
+           return 0;
+         },
+         aliasAnalysisFromSchema()),
+     Operator(
+         "prim::T(Tensor(a) a) -> Tensor(a)",
+         [](Stack& stack) {
+           at::Tensor a;
+           pop(stack, a);
+           push(stack, a.numpy_T());
+           return 0;
+         },
+         aliasAnalysisFromSchema()),
+     Operator(
+         "prim::output_nr(Tensor a) -> int",
+         [](Stack& stack) {
+           at::Tensor a;
+           pop(stack, a);
+           push(stack, a.output_nr());
+           return 0;
+         },
+         aliasAnalysisFromSchema()),
+     Operator(
+         "prim::ndim(Tensor a) -> int",
```

Comment:
This is the same op which already exists as `aten::dim`, and which we have optimizations for https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/peephole.cpp#L363.  It would be nice if passes did not have to reason about duplicate passes & we didn't register an additional op that we then couldn't remove because of BC. 

Could you add a `ndim` function [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/frontend/builtin_functions.cpp#L40) under the `prim` namespace?
```
def ndim(x):
    return x.dim()
```